openwisp_monitoring/device/apps.py

```diff
@@ -11,3 +15,57 @@ class DeviceMonitoringConfig(AppConfig):
 
     def ready(self):
         manage_short_retention_policy()
+        from openwisp_controller.connection.models import DeviceConnection
+        from openwisp_controller.connection.signals import is_working_changed
+        is_working_changed.connect(self.health_status_changed,
+                                   sender=DeviceConnection,
+                                   dispatch_uid='change_health_status')
+
+    @classmethod
+    def health_status_changed(cls, instance, is_working, **kwargs):
+        from ..check.models import Check
+        d = instance.device
+        dm = d.monitoring
+        initial_status = dm.status
+        status = 'ok' if is_working else 'problem'
+        if not is_working:
+            dm.update_status(status)
+            # Create a related notification explaining why it's not working
+            desc = instance.failure_reason
+            opts = dict(actor=d, level='warning', verb='not working', description=desc)
+            if initial_status == 'ok':
+                try:
+                    # if checks exist trigger them
+                    Check.objects.get(object_id=d.pk)
+                    management.call_command('run_checks')
```

Comment:
will execute all checks, we must execute only checks of the device

Reply:
@nemesisdesign, I will update this after PR`#22` as there I have created a `find_related_checks` task for the same purpose. :smile: